text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,419 @@
+W3C `sampled` flag is set, which determines whether child contexts
+will be sampled by a `ParentBased` Sampler.  It is useful to know the
+head trace sampling probability associated with a context in order to
+build span-to-metrics pipelines when the built-in `ParentBased`
+Sampler is used.  Further motivation for supporting span-to-metrics
+pipelines is presented in [OTEP
+170](https://github.com/open-telemetry/oteps/pull/170).
+
+A consistent trace sampling decision is one that can be carried out at
+any node in a trace, which supports collecting partial traces.
+OpenTelemetry specifies a built-in `TraceIDRatioBased` Sampler that
+aims to accomplish this goal but was left incomplete (see a
+[TODO](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#traceidratiobased) 
+in the v1.0 Trace specification).
+
+We propose to propagate the necessary information alongside the [W3C
+sampled flag](https://www.w3.org/TR/trace-context/#sampled-flag) using
+`tracestate` with an `otel` vendor tag, which will require
+(separately) [specifying how the OpenTelemetry project uses
+`tracestate` itself](https://github.com/open-telemetry/opentelemetry-specification/pull/1852).
+
+## Explanation
+
+Two pieces of information are needed to convey consistent head trace
+sampling probability:
+
+1. p-value representing the head trace sampling probability.
+2. r-value representing the "randomness" as the source of consistent sampling decisions.
+
+This proposal uses 6 bits of information for each of these and does
+not depend on built-in TraceID randomness, which is not sufficiently
+specified for probability sampling at this time.  This proposal closely 
+follows [research by Otmar Ertl](https://arxiv.org/pdf/2107.07703.pdf).
+
+### p-value
+
+To limit the cost of this extension and for statistical reasons
+documented below, we propose to limit head trace sampling probability
+to powers of two.  This limits the available head trace sampling
+probabilities to 1/2, 1/4, 1/8, and so on.  We can compactly encode
+these probabilities as small integer values using the base-2 logarithm
+of the adjusted count.
+
+Using six bits of information we can convey unknown and known sampling
+rates as small as 2**-61.  The value 63 is reserved to mean sampling
+with probability 0, which conveys an adjusted count of 0 for the
+associated context.
+
+When propagated, the "p-value" as it is known will be interpreted as
+shown in the folowing table.  The p-value for known sampling
+probabilities is the negative base-2 logarithm of the probability,
+offset by +1 to so that the 0 p-value can be treated as unknown (for
+backwards compatibility):
+
+| p-value | Head Probability | Note                                                 |
+| -----   | -----------      | ----                                                 |
+| 0       | Unknown          | Do not propagate `p=0`, instead omit from tracestate |
```

Comment:
Non-blocking comment for @jsuereth: Note there is a question about field presence here. If OTel adopts the use of proto3 field presence then we can distinguish between 0 and "not present", that would change (and simplify) the backwards compatibility story. Then p-values would equal the effective `s` value and the invariant would be written `sampled <=> p <= r`.